src/modules/tensor/cpu/kernel/jpeg_compression_distortion.cpp

```diff
@@ -0,0 +1,2334 @@
+{
+    quality = std::max(1, std::min(quality,100));
+    Rpp32f qualityFactor;
+    if (quality < 50)
+        qualityFactor = 50.0f / quality;
+    else
+        qualityFactor = 2.0f - (quality / 50.0f);
+    return qualityFactor;
+}
+
+inline void upsample_avx2(__m256 input, __m256 *p)
+{
+    // Unpack and duplicate values to achieve upsampling
+    p[0] = _mm256_permutevar8x32_ps(input, _mm256_setr_epi32(0, 0, 1, 1, 2, 2, 3, 3));
+    p[1] = _mm256_permutevar8x32_ps(input, _mm256_setr_epi32(4, 4, 5, 5, 6, 6, 7, 7));
+}
+
+inline void transpose_8x8_avx(__m256* p)
+{
+    __m256 temp[16];
+
+    temp[0] = _mm256_unpacklo_ps(p[0], p[2]);
+    temp[1] = _mm256_unpackhi_ps(p[0], p[2]);
+    temp[2] = _mm256_unpacklo_ps(p[1], p[3]);
+    temp[3] = _mm256_unpackhi_ps(p[1], p[3]);
+    temp[4] = _mm256_unpacklo_ps(p[4], p[6]);
+    temp[5] = _mm256_unpackhi_ps(p[4], p[6]);
+    temp[6] = _mm256_unpacklo_ps(p[5], p[7]);
+    temp[7] = _mm256_unpackhi_ps(p[5], p[7]);
+
+    temp[8]  = _mm256_unpacklo_ps(temp[0], temp[2]);
+    temp[9]  = _mm256_unpackhi_ps(temp[0], temp[2]);
+    temp[10] = _mm256_unpacklo_ps(temp[1], temp[3]);
+    temp[11] = _mm256_unpackhi_ps(temp[1], temp[3]);
+    temp[12] = _mm256_unpacklo_ps(temp[4], temp[6]);
+    temp[13] = _mm256_unpackhi_ps(temp[4], temp[6]);
+    temp[14] = _mm256_unpacklo_ps(temp[5], temp[7]);
+    temp[15] = _mm256_unpackhi_ps(temp[5], temp[7]);
+
+    p[0] = _mm256_permute2f128_ps(temp[8], temp[12], (2 << 4) | 0);
+    p[4] = _mm256_permute2f128_ps(temp[8], temp[12], (3 << 4) | 1);
+    p[1] = _mm256_permute2f128_ps(temp[9], temp[13], (2 << 4) | 0);
+    p[5] = _mm256_permute2f128_ps(temp[9], temp[13], (3 << 4) | 1);
+    p[2] = _mm256_permute2f128_ps(temp[10], temp[14], (2 << 4) | 0);
+    p[6] = _mm256_permute2f128_ps(temp[10], temp[14], (3 << 4) | 1);
+    p[3] = _mm256_permute2f128_ps(temp[11], temp[15], (2 << 4) | 0);
+    p[7] = _mm256_permute2f128_ps(temp[11], temp[15], (3 << 4) | 1);
+}
+
+inline void quantize_block(Rpp32f *block, const Rpp32f *quantTable, Rpp32s stride, Rpp32s qualityParam)
+{
+    Rpp32f qualityFactor = get_quality_factor(qualityParam);
+    for (Rpp32s row = 0; row < 8; row++)
+    {
+        Rpp32s rowIdx = row * stride;
+        Rpp32s rowQuantIdx = row * 8;
+        for (Rpp32s col = 0; col < 8; col++)
+        {
+            Rpp32f qCoeff = quantTable[rowQuantIdx + col] * qualityFactor; // Directly accessing 1D array
+            qCoeff = (qCoeff < 1.0f) ? 1.0f : ((qCoeff > 255.0f) ? 255.0f : qCoeff);
+            block[rowIdx + col] = qCoeff * roundf(block[rowIdx + col] / qCoeff);
+        }
+    }
+}
+
+template <typename T>
+inline void dct_8x8_1d_core(T *x)
+{
+    T temp[12];
+
+    temp[0] = x[0] + x[7];
+    temp[1] = x[1] + x[6];
+    temp[2] = x[2] + x[5];
+    temp[3] = x[3] + x[4];
+
+    temp[4] = x[0] - x[7];
+    temp[5] = x[6] - x[1];
+    temp[6] = x[2] - x[5];
+    temp[7] = x[4] - x[3];
+
+    temp[8] = temp[0] + temp[3];
```

Comment:
please combine temp[0] and temp[3] computation here and use only T temp[4] like
temp[8] = x[0] + x[7] + x[3] + x[4];